server/@types/payloads/types.d.ts

```diff
@@ -0,0 +1,41 @@
+declare module 'payloads' {
+    interface LectureSubPayload {
+        event: string,
+        data: {
+            lecture_id: string
+        }
+    }
+
+    interface StudentSubPayload {
+        event: string,
+        data: {
+            student_id: string,
+            lecture_link: string
+        }
+    }
+
+    interface QuizRequestPayload {
+        event: string,
+        data: {
+            lecture_id: string,
+            student_ids: string[],
+            questions: any
+        }
+    }
+
+    interface QuizResponsePayload {
+        event: string,
+        data: {
+            lecture_link: string,
+            student_id: string,
+            answers: any
+        }
+    }
```

Comment:
Maybe use a common interface named for ex: `SocketMessage` with event and data field, that Payloads implements.